components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -99,6 +106,10 @@ class SitePermissionsFeature(
     val onShouldShowRequestPermissionRationale: (permission: String) -> Boolean,
     private val store: BrowserStore
 ) : LifecycleAwareFeature, PermissionsFeature {
+    @VisibleForTesting
+    internal val selectOrAddUseCase by lazy {
+        SelectOrAddUseCase(store)
+    }
```

Comment:
nit: Maybe apps could pass a reference of the use case?

Reply:
Thought about this in the scenario that the the usecase is being passed in already consatructed with a specific store.
But since the feature receives the store as an argument and used it already to search for tabs or observe various changes it seems like the functionality is already tied to this specific store.
Then I thought that we could `hoist` the event of the user clicking the "Learn more" link and let the user decide what should happen - maybe use a different usecase, which would've also helped with the above scenario of letting clients control the documentation URL but this seemed like a bit too bigger change for just one permission from the many that the feature helps with.

Reply:
👍🏽